src/Struct/FindologicProduct.php

```diff
@@ -762,14 +762,30 @@ protected function setOrdernumberByProduct(ProductEntity $product): void
     protected function setCategoriesAndCatUrls(): void
     {
         $productCategories = $this->product->getCategories();
-        if ($productCategories === null || empty($productCategories->count())) {
+        $productVariants = $this->product->getChildren();
+
+        // If there are no product categories, and no children either, we throw ProductHasNoCategoriesException.
+        if ($productCategories->count() === 0 && $productVariants->count() === 0) {
             throw new ProductHasNoCategoriesException($this->product);
         }
 
         $catUrls = [];
         $categories = [];
 
         $this->parseCategoryAttributes($productCategories->getElements(), $catUrls, $categories);
+
+        if ($productVariants) {
```

Comment:
This thuthiness will always pass, as an empty `ProductCollection` instance is never considered falsy. Only if it was `null`, this wouldn't pass, however we set the proper association in the `ProductService` already, therefore this will never be `null`. Consider `$productVariants->count() > 0`.